evalml/tests/pipeline_tests/test_time_series_pipeline.py

```diff
@@ -1724,3 +1724,76 @@ def test_drop_time_index_woodwork(ts_data, time_series_regression_pipeline_class
 
     assert isinstance(X_t.index, pd.DatetimeIndex)
     assert isinstance(y_t.index, pd.DatetimeIndex)
+
+
+@pytest.mark.parametrize(
+    "forecast_horizon,gap,max_delay",
+    [(1, 0, 1), (1, 1, 2), (2, 0, 2), (3, 1, 2), (1, 2, 2), (2, 7, 3), (3, 2, 4)],
+    # [(1, 0, 1)],
+)
+@patch("evalml.pipelines.components.RandomForestRegressor.fit")
+@patch("evalml.pipelines.components.RandomForestRegressor.predict")
+def test_dates_needed_for_prediction(
+    mock_predict,
+    mock_fit,
+    forecast_horizon,
+    gap,
+    max_delay,
+    ts_data,
+    time_series_regression_pipeline_class,
+):
+    X, X_t, y = ts_data()
```

Comment:
Based on Frank's comment above, it'd probably be a good idea to test a few different frequencies here!